Link Medley releases from Interlisp.org

pamoroso opened the pull request:
This patch adds to the [News and Status Reports](https://interlisp.org/project/status) page the section Release Notes to link to the [releases page](https://github.com/Interlisp/medley/releases) of Medley's GitHub repo. It addresses [issue #1681](https://github.com/Interlisp/medley/issues/1681).

pamoroso committed:
Link Medley releases from News & Status Reports

Add section Release Notes to link to the releases section of Medley's GitHub repo.

masinter commented:
aat last Monday's meeting we decided to change the weekday to Tuesday (2 AM). That's because the implementor's group meets on Monday and we often resolve open PRs then.

These are untested

masinter reviewed (approved):
Perhaps just saying "built" because "released" implies more QA than we are doing.

pamoroso committed:
Release -> automatic build, remove day reference

Replace references to releases with automatic builds. Don't mention the specific day of the weekly builds.

pamoroso commented:
I rephrased to replace the references to releases with builds, adding that the builds are automatic to de-emphasize QA. I also removed the specific weekday of the builds.
